### ISSUE TITLE
Memory pressure when loading swagger.json

### ISSUE THREAD

**Issue by Blackbaud-ChristiSchneider** [domaindrivendev/Swashbuckle.AspNetCore#1950]
When I start my web service which produces a swagger document using Swashbuckle, I note the memory pressure and let it stabilize. When I load the swagger document, the memory pressure rises by a significant amount and never falls. This does not appear to grow if the swagger document is reloaded, so doesn't seem to be leaking memory, just never releasing it.

I'm still trying to diagnose this and am going to attempt to create a sample project to reproduce but if anyone can check it themselves and confirm if they see the same results, that would be very helpful.

I'm using the latest `Swashbuckle.AspNetCore`, `Swashbuckle.AspNetCore.Annotations`, `Swashbuckle.AspNetCore.Filters`, and `Swashbuckle.AspNetCore.Newtonsoft` nuget versions and .net core 3.1. I personally am using a Mac but we first noticed the issue on our production server which runs Linux and confirmed with a coworker that they see similar memory pressure on their Windows dev machine.

I've removed all of the custom swagger code in my web service and verified that this still occurs, although the memory increase is smaller. In my tests, the service has a baseline of around 65 MB of memory. When I load the swagger document with no changes, the memory goes up to around 200 MB. When I remove all of my custom code and load the swagger document, the memory goes up to around 110 MB.

**Comment by Blackbaud-ChristiSchneider** [comment]
I am able to reproduce this with the Basic.csproj test project included in this repository. I changed the project settings so that it does not open the browser when it starts up and checked the memory before loading the swagger - it's around 25 MB. After loading the swagger.json url (http://localhost:5000/swagger/v1/swagger.json) (I'm eliminating the Swagger UI as a variable by just loading the json) the memory jumps to around 35 MB and does not drop. With this project, I am actually seeing subsequent reloads increase the memory by around 3 MB per reload and not drop, so maybe there is a leak.

**Comment by domaindrivendev** [comment]
`ApiExplorer`, the built-in API metadata component that's part of ASP.NET Core and is heavily utilized by Swashbuckle performs read-through caching of it's data, hence the initial memory increase on first load of the Swagger JSON. See the following related issues for more info on this and how it can be problematic when your API project uses extremely DTO types.

#1864 
#1894 

Re the possibility of an actual leak, could you please share some screeshots of the profiling you've been doing to demonstrate this? Have you accounted for Garbage Collection (or lack of) in your tests?

I know the Swashbuckle code intimately and although I can't rule it out, I can't think of any places where it could be causing a memory leak. If there is one, it's possible it could be happening within `ApiExplorer`, in which case we'd need to prove and then create an upstream issue in the ASP.NET Core repo.

**Comment by Blackbaud-ChristiSchneider** [comment]
I've been struggling with the memory profiler tools on mac, the GC always reads zero even when there is an actual memory leak. I think I'll have to use a windows machine to diagnose if there's an actual problem.

**Comment by bdshanto** [comment]
Every reload increase memory 5-8 MB. 
We are struggling of these issues. Finding a way to resolve these issues, otherwise we have to move, because these make performance issue 👎

**Comment by bdshanto** [comment]
> Every reload increase memory `5-8 MB`. We are struggling of these issues. Finding a way to resolve these issues, otherwise we have to move, because these make performance issue 👎

**Comment by kevinhlyman** [comment]
Basically the same thing is happening to us. We also first noticed it in our production service in Azure. We initially thought it was an Azure issue and were working that. But I've since moved to my work development machine and used VS to watch the memory usage. What I've noticed is that with each call to swagger.json the memory jumps up about 5-8 MB like @bdshanto also notes. Then some GC eventually happens and the memory levels off. If I keep calling the swagger.json, hitting F5, it will eventually do a big memory jump with GC. It will jump 100-200 MB and it never lets it's go.
![swaggerjsonmemoryjump](https://user-images.githubusercontent.com/7108255/219488826-2e9f11eb-7dc8-42bf-9ae8-3cef5fc72fff.PNG)
